src/engine/strat_engine/blockdev.rs

```diff
@@ -7,33 +7,44 @@
 use std::fs::OpenOptions;
 use std::path::PathBuf;
 
-use chrono::{DateTime, Utc};
+use chrono::{DateTime, TimeZone, Utc};
 
 use devicemapper::{Device, Sectors};
 
+use super::super::engine::{BlockDev, HasUuid};
 use super::super::errors::EngineResult;
-use super::super::types::{DevUuid, PoolUuid};
+use super::super::types::{BlockDevState, DevUuid, PoolUuid};
 
 use super::metadata::BDA;
 use super::range_alloc::RangeAllocator;
 use super::serde_structs::{BlockDevSave, Recordable};
 
 
 #[derive(Debug)]
-pub struct BlockDev {
+pub struct StratBlockDev {
     dev: Device,
     pub devnode: PathBuf,
     bda: BDA,
     used: RangeAllocator,
+    pub location: Option<String>,
```

Comment:
```location``` is probably a bad name for this field and ```user_id``` might not be so good for accessor method. How about ```user_supplied_string```... that's descriptive.

Reply:
Well, we (meaning..me) has let a little inconsistency arise between the design doc and the DBus API doc. Design doc calls them `location` and `disk_id`, API doc calls them `UserId` and `HardwareId`. I'm using the design doc terms here. Would you be in favor of adopting the API doc's terms in the design doc? Or some other terms. Consistency is good. Naming is hard. 🙁